### PR TITLE
Change model + barman update with associations, functionnal

### PR DESCRIPTION
**Pull request by kistora**
### Description of your PR

Implement barman update with association changes. I had to change the back (add associations to `GET /api/barmen/{id}`) in order to implement it.

### Pre-review TODO

- [x] PR is rebased on top of `origin/master` 
- [ ] PR provide new tests for new behaviors
- [x] PR doesn't introduce commented out code
- [x] PR has been tested locally on the main use cases
- [ ] Documentation have been updated if necessary

### CODE REVIEW

File: client/src/app/barmen/new/barman-new.component.html [hunk 1]
```diff
@@ -30,19 +30,18 @@ <h1>Ajouter un Barman</h1>
                     Surnom <strong>requis</strong>
                     </mat-error>
                 </mat-form-field>
-                <mat-form-field>
+                <!-- <mat-form-field>
```
Comment (Embraser01) [inline client/src/app/barmen/new/barman-new.component.html]: Why comment this part ?
Back should create a new barman with an username (and password also but this is not in the spec I think)

File: client/src/app/barmen/new/barman-new.component.ts [hunk 2]
```diff
@@ -34,7 +34,7 @@ export class BarmanNewComponent {
         barman.firstName = this.firstName;
         barman.lastName = this.lastName;
         barman.nickname = this.nickname;
-        barman.username = this.username;
+        // barman.username = this.username;
```
Comment (Embraser01) [inline client/src/app/barmen/new/barman-new.component.ts]: Same here ?

File: client/src/app/barmen/new/barman-new.component.html (hunk 1 above)
Comment (kistora) [reply client/src/app/barmen/new/barman-new.component.html]: I would implement it in an other PR. You are right, I'm going to restore this file.

File: client/src/app/barmen/new/barman-new.component.ts (hunk 2 above)
Comment (kistora) [reply client/src/app/barmen/new/barman-new.component.ts]: Same

**Pull request merged**